webapp/src/components/ExchangePage/index.jsx

```diff
@@ -369,112 +438,239 @@ function getMarketRate(
-                continue
-              }
-              const vault = await uniswapEXContract.vaultOfOrder(...Object.values(order))
-              const amount = await new Promise((resolve, reject) =>
-                window.web3.eth.call(
-                  {
-                    to: order.fromToken,
-                    data: `${BALANCE_SELECTOR}000000000000000000000000${vault.replace('0x', '')}`
-                  },
-                  (error, amount) => {
-                    if (error) {
-                      reject(error)
-                    }
-                    resolve(amount)
-                  }
-                )
-              )
-              if (order && ordersAdded[orderData] === undefined) {
-                orders.push({ ...order, amount })
-                ordersAdded[orderData] = orders.length - 1
-              }
-            } catch (e) {
-              console.warn(`Error decoding order: ${orderData}`)
-            }
-          }
+function findOrders(data) {
+  const orders = []
+  const transfers = data.split('a9059cbb').slice(1)
+
+  for (const transfer of transfers) {
+    if (transfer.length >= 704) {
+      const order = transfer.slice(256, 706)
+      // Uniswap orders have a pk and address embeded on the tx
+      // we look for those
+      const pk = `0x${order.slice(320, 320 + 64)}`
+      const expectedAddress = `0x${order.slice(408, 408 + 40)}`
+      if (new ethers.Wallet(pk).address.toLowerCase() === expectedAddress) {
+        orders.push(`0x${order.slice(1)}`)
+      }
+    }
+  }
+
+  return orders
+}
+
+async function backfillOrders(account, onUpdate = _ => {}) {
+  const reviewedTxs = new Set()
+
+  const targetBlock = await readWeb3.eth.getBlockNumber()
+  const last = getLastBackfill(account)
+  console.info(`Running backfill for ${account} - ${last}/${targetBlock}`)
+
+  const logs = await readWeb3.eth.getPastLogs({
+    fromBlock: last,
+    toBlock: targetBlock,
+    topics: [
+      '0xddf252ad1be2c89b69c2b068fc378daa952ba7f163c4a11628f55a4df523b3ef', // Transfer topic
+      `0x000000000000000000000000${account.replace('0x', '')}` // Account topic
+    ]
+  })
+
+  console.info(`Found ${logs.length} candidates for ${account}`)
+
+  let bufferEval = 0
+  for (const log of logs) {
+    const txHash = log.transactionHash
+    if (!reviewedTxs.has(txHash)) {
+      reviewedTxs.add(txHash)
+      const tx = await readWeb3.eth.getTransaction(txHash)
+      const orders = findOrders(tx.input)
+
+      for (const order of orders) {
+        console.info(`Found order ${txHash}`)
+        saveOrder(account, order)
+        onUpdate(log.blockNumber)
+      }
+
+      bufferEval++
+      if (bufferEval > 10) {
+        setLastBackfill(account, log.blockNumber)
+        bufferEval = 0
+      }
+    }
+  }
+
+  setLastBackfill(account, targetBlock)
+  console.info(`Finished backfill for ${account} up to ${targetBlock}`)
+  return targetBlock
+}
+
+async function backfillEthOrders(account, uniswapEx) {
```

Comment:
Why the update method is not necessary here?

Reply:
Because the ETH backfill is a quick process that should only take 1 second, it's only retrieving the logs and the orders are already encoded there, the token backfill can take almost 1 minute, so the `update` method can trigger in-between renders